examples/chip-tool/commands/common/CHIPCommand.cpp

```diff
@@ -48,6 +51,47 @@ const chip::Credentials::AttestationTrustStore * CHIPCommand::sTrustStore = null
 chip::Credentials::GroupDataProviderImpl CHIPCommand::sGroupDataProvider{ kMaxGroupsPerFabric, kMaxGroupKeysPerFabric };
 
 namespace {
+
+class StringCommaSplitter
+{
+public:
+    StringCommaSplitter(const char * s) : mData(strdup(s))
+    {
+        VerifyOrDie(mData != nullptr);
+        mNext = mData;
+    }
+    ~StringCommaSplitter() { free(mData); }
+
+    // returns null when no more items available
+    const char * Next()
+    {
+        char * current = mNext;
+        if (*current == '\0')
+        {
+            return nullptr; // nothing left
+        }
+
+        char * comma = strchr(current, ',');
+
+        if (comma != nullptr)
+        {
+            mNext  = comma + 1;
+            *comma = '\0';
+        }
+        else
+        {
+            // last element, position on the final 0
+            mNext = current + strlen(current);
+        }
+
+        return current;
+    }
+
+private:
+    char * mNext; // next element to return by calling Next()
+    char * mData; // allocated data
+};
```

Comment:
`chip-tool` use to do all the argument parsing in `examples/chip-tool/commands/common/Command.cpp`

While there is no support for comma separated strings there is already some support for comma separated numbers, bools.
You probably want to do something similar.

Reply:
Will take a look. For now started to split the comma iteration in #27154

Reply:
Updated to leave the parsing up to command.